src/common/utils.ts

```diff
@@ -14,6 +14,91 @@ export function utf8Decode(data: Uint8Array): string | undefined {
   }
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
+function parseAlgosdkV2SimulateResponse(obj: any): any {
```

Comment:
Please note, `SimulateResponse` objects encoded via `JSON.stringify(simulateResponse.get_obj_for_encoding())` using <.3.0 `js-algorand-sdk` results in a json string where addresses and byte fields are encoded into base64. While >3.0 `js-algorand-sdk` clients will fail when attempting to `algosdk.decodeJSON(jsonEncodedSimualteResponseString, SimulateResponse)` -> since it doesn't expect base64 strings for address fields and keys like gh and apaa. So most of the compatibility issues seems to be on the encoded transaction objects. Wasn't clear if v2 encoded SimulateResponses are compatible with v3 decoders based on the note on related topic in migration guide https://github.com/algorand/js-algorand-sdk/blob/develop/v2_TO_v3_MIGRATION_GUIDE.md#object-encoding-and-decoding. 

If it's a breaking change and its expected behaviour on js sdk side, we will probably keep the implementation here as is to have backwards compatibility with projects that encoded simulate traces with v2 sdk. Otherwise I can submit it as issue on js-sdk repo and keep this method temporarily until resolved on js-sdk 👍

Reply:
You're right that algosdk v2 would produce base64-encoded raw address bytes when asked to convert a simulate response to JSON like that. I consider that a bug in v2, since that's not the way algod produces a JSON simulate response.

I believe simulate responses are the only model type that's affected by this issue, since that's the only response type in v2 which unpacks into a typed response class and which comes from a msgpack-encoded response (meaning the addresses are internally stored as byte arrays, not strings).

While this has been fixed in v3, you're correct in pointing out that there's a potential compatibility issue if you're using an artifact that was produced using the v2 SDK.

From my perspective, adding code to this repo which handles compatibility between the two versions seems like a reasonable approach.